lesson18/src/js/main.js

```diff
@@ -0,0 +1,299 @@
+  const activeImage = document.querySelector(".is-active");
+  const currentIndex = Number(activeImage.dataset.slideIndex);
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+  prevButton.disabled = currentIndex === firstIndex;
+  nextButton.disabled = currentIndex === lastIndex;
+}
+
+function renderSlidContents(slideImageData){
+  renderSlideArea();
+  renderSlideImage(slideImageData);
+  renderPagenation(slideImageData.length);
+  renderPagenationBullet(slideImageData.length);
+  renderSlideButton();
+  addSwitchButtonEvent();
+  clickPagenationBullet();
+  toggleButtonDisabled();
+  autoPlayslide();
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(
+        fetchImageData(
+          //https://api.json-generator.com/templates/IZjWl012CAMD/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl : image = 0
+          //https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&status=503 : 503
+          "https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl"
+        )
+      );
+    }, 3000);
+  });
+}
+
+async function fetchImageData(URL) {
+  const response = await fetch(URL);
+  if (response.ok) {
+    const json = await response.json();
+    return json;
+  } else {
+    console.error(`${response.status}:${response.statusText}`);
+    displayErrorMassage("サーバの通信に失敗しました");
+  }
+}
+
+async function getSlideImage(){
+  showLoadingImage();
+  try{
+     const json = await callImageData();
+     const slideData = json.slide;
+    if(slideData.length > 0){
+      renderSlidContents(slideData);
+    }
+  }catch{
+    addNoimage();
+    console.log('表示する画像がありませんでした');
+  }finally{
+    removeLoading();
+  }
+}
+
+async function init() {
+  const slideContents = await getSlideImage();
+  slideContents && autoPlayslide();
+}
+
+init();
+
+
+
+function addSwitchButtonEvent(){
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+  prevButton.addEventListener("click", (e) => {
+    switchPagination(e.currentTarget);
+    switchSlider("previousElementSibling");
+  });
+  nextButton.addEventListener("click", (e) => {
+    switchPagination(e.currentTarget);
+    switchSlider("nextElementSibling");
+  });
+}
+
+function switchSlider(slideTarget) {
+  switchImg(slideTarget);
+  switcPagenationBullet(slideTarget);
+  toggleButtonDisabled();
+}
+
+let autoPlayID;
+function autoPlayslide() {
+  autoPlayID = setInterval(() => {
+    const pagenationCurrent = document.getElementById("js-current");
+    let currentNum = Number(pagenationCurrent.textContent);
+    if(currentNum === 5){
```

Comment:
IMO
Magic numbers should be avoided.

Reply:
With this, it will not work if there are 5 or more images.
Corrected to judge by the number of images instead of the fixed value.

[5245424](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/52454246ff65202451054115fc34b364ffc8396f)